gtsam/sfm/ShonanAveraging.cpp

```diff
@@ -470,8 +471,54 @@ Sparse ShonanAveraging<d>::computeA(const Values &values) const {
   return Lambda - Q_;
 }
 
-/* ************************************************************************* */
-/// MINIMUM EIGENVALUE COMPUTATIONS
+// Alg.6 from paper Distributed Certifiably Correct Pose-Graph Optimization
+static bool PowerMinimumEigenValue(
```

Comment:
You only actually need S here (check Alg 6 in the paper)

Reply:
@jingwuOUO please reply. I am also not clear on why we need S.

Reply:
Sorry for the misunderstanding here. The S has different meanings in two paper. So in the code, S is used here to provide the initial vector x0 in power method.

(1) In the Yulun20 paper, S = S(X), here S is A in the code. X is actually S in the code.

(2) In our ECCV paper, A in the code is the certificate matrix (which is S in (1)). S in the code is S \in St(d, p)^n in our paper.

Reply:
OK, so just to clarify then:

- A is the matrix whose eigenpair we want to compute
- S is meant to be the first-order-critical Stiefel manifold element found by local search in the Riemannian Staircase in Shonan Averaging.  We only actually use this for constructing a good starting vector x0 for the power method, so we can get a cleaner (and more general) interface by replacing S with an (optional) argument of the form ```const Vector& x0```, which provides an (optional) initialization for the power method; if the user doesn't supply this argument, we can just take a randomly-sampled vector of appropriate dimension.